bcipy/task/paradigm/rsvp/calibration/calibration.py

```diff
@@ -166,31 +192,46 @@ def execute(self):
         self.rsvp.draw_static()
         self.window.flip()
 
-        # Give the system time to process
-        core.wait(self.buffer_val)
-
-        # Write offset
-        if self.daq.is_calibrated:
-            _write_triggers_from_inquiry_calibration(
-                ['offset',
-                 self.daq.offset(self.rsvp.first_stim_time)],
-                self.trigger_file,
-                offset=True)
-
-        # Close this sessions trigger file and return some data
-        self.trigger_file.close()
-
         # Wait some time before exiting so there is trailing eeg data saved
         core.wait(self.eeg_buffer)
 
         return self.file_save
 
+    def write_trigger_data(self, timing: List[Tuple[str, float]], first_run) -> None:
+        # write offsets. currently, we only check for offsets at the beginning.
+        if self.daq.is_calibrated and first_run:
+            self.trigger_handler.add_triggers(
+                [Trigger(
+                    'starting_offset',
+                    TriggerType.OFFSET,
+                    # offset will factor in true offset and time relative from beginning
+                    (self.daq.offset(self.rsvp.first_stim_time) - self.rsvp.first_stim_time)
+                )]
+            )
+
+        # make sure triggers are written for the inquiry
+        self.trigger_handler.add_triggers(self.convert_timing_triggers(timing))
+
+    def write_offset_trigger(self) -> None:
```

Comment:
Is this method used? It looks like it was replaced with `write_trigger_data`.

Reply:
They are both used! I still write the offset only value in case someone did want the LSL timestamps... not sure if this is helpful or not. It's filtered out as a TriggerType.SYSTEM when it loads by default.